fixes for s360 errors under a new api version

vasundhraManhas opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Commit:
fixes for s360 errors under a new api version

Commit:
updating version in v1 swagger

Commit:
updating readme for 2022-01-30-preview version

Commit:
suppress secret property and add directives

Commit:
suppressing R4016 and R4017

Commit:
prettier check

pilor commented:
Please publish this PR (take it out of draft mode) when its ready and @ me for an ARM API review

zhenglaizhang commented:
@vasundhraManhas your PR contains cross version breaking change https://github.com/Azure/azure-rest-api-specs/pull/17190/checks?check_run_id=4666962430 and need do breaking change review.

vasundhraManhas commented:
> @vasundhraManhas your PR contains cross version breaking change https://github.com/Azure/azure-rest-api-specs/pull/17190/checks?check_run_id=4666962430 and need do breaking change review.

@zhenglaizhang 
the following are the 3 errors in cross-version problem 

a) for the 1005-RemovedPath error, we have removed the mentioned operations API from datamigration.json (which was the latest stable swagger) but we have added this path in the sqlmigration.json (the new preview version) and we will make changes so that operations api request always goes to the new endpoint and returns both the old and new operations. Since we are keeping both datamigration.json and sqlmigration.json in the 2022-01-30-preview folder, will that be fine or is it still a breaking change problem?



b) for the 1006-RemovedDefinition error, there is no information being provided regarding the error . clicking on the provided links opens a page not found tab . Any suggestions here?



c) for the 1007-RemovedClientParameter error, again the referred parameter was removed from datamigration.json but it is there in sqlmigration.json and datamigration.json is referring to this parameter from sqlmigration.json (something like this : "$ref": "./sqlmigration.json#/parameters/SubscriptionIdParameter" ) . So, should this be fine too ?


We had these exact same issues in our previous PR (for 2021-10-30-preview version) and they got a pass without the breaking change review process. Here's the PR:  https://github.com/Azure/azure-rest-api-specs/pull/16032/checks?check_run_id=4555175737

vasundhraManhas commented:
> Please publish this PR (take it out of draft mode) when its ready and @ me for an ARM API review

@pilor  Hi, the PR is ready for review

ruowan commented:
/azp run

ruowan commented:
/azp run

pilor commented:
This guidance was not followed (from PR description):  "To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits."

Please rebase to accomplish that or point out which files have actual changes vs. the previous API version

vasundhraManhas commented:
> This guidance was not followed (from PR description): "To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits."
> 
> Please rebase to accomplish that or point out which files have actual changes vs. the previous API version

@pilor  Hi, these are the actual files changed:-
1) Common.json
2) Projects.json
3) Services.json
4) Tasks.json
5) MigrateSchemaSqlServerSqlDbTask.json
6) MigrateSqlServerSqlDbTask.json
7) MigrateSqlServerSqlMITask.json
8) TasksCommon.json
9) ConnectToSourceSqlServerTask.json
10) MigratePostgreSqlAzureDbForPostgreSqlSyncTask.json

A list of properties added/deleted/modified in these files can be found in the table here : https://msdata.visualstudio.com/Database%20Systems/_workitems/edit/1564885

pilor commented on specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2022-01-30-preview/definitions/Projects.json:
```diff
@@ -0,0 +1,163 @@
+        },
+        "azureAuthenticationInfo": {
+          "description": "Field that defines the Azure active directory application info, used to connect to the target Azure resource",
+          "type": "object"
```
>object [](http://example.com/codeflow?start=19&length=6)

Why is this an anonymous object with no schema?

vasundhraManhas replied:
added now

pilor commented on specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2022-01-30-preview/definitions/Tasks.json:
```diff
@@ -0,0 +1,1017 @@
+          },
+          "readOnly": true
+        },
+        "createdOn": {
```
>createdOn [](http://example.com/codeflow?start=9&length=9)

this should be coming from the systemData common contract on this resource.  If this is an s360 fix since you already have this property its fine.  If its a new property please do not add it

pilor replied:
applies to the other 2 places this property was added

vasundhraManhas replied:
This was added because this was an s360 error .  I have now added systemData instead. System data has "createdAt" instead of "createdOn" , will this still resolve the s360 error?

pilor replied:
This is not where systemData lives in the resource.  The resources already have systemData at the correct place on line 132.  I can't help you decide what will/won't fix your S360 errors.  However, if this is a brand new property you aren't exposing today then do not add it.  If its something that is already exposed in your API versions (but not documented in swagger) then keep it

vasundhraManhas replied:
Alright, "createdOn" was  already there and is not a brand new property so we can keep it

Commit:
adding systemData property

Commit:
adding schema for azureAuthenticationInfo

Commit:
Revert "adding systemData property"

This reverts commit a0e3edef990325243127e1b51c7ce1e6c4ef905a.

Commit:
nit error

zhenglaizhang commented:
> 

I checked with breaking change owner and removing path needs to do review, you can present to breaking change review board to get approval

Commit:
API description changes

Commit:
adding get examples with expand parameter

Commit:
prettier check

Commit:
using the newly added get examples (with expand parameter)

Commit:
nit

vasundhraManhas commented:
/azp run

Commit:
prettier check

Commit:
Solving new s360 error of invalid type in dictonary

Commit:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into s360

# Conflicts:
#	package-lock.json

Commit:
resolving OBJECT_ADDITIONAL_PROPERTIES type of errors by adding the missing properties

Commit:
Resolving ENUM_MISMATCH and OBJECT_MISSING_REQUIRED_PROPERTY errors

Commit:
nit: adding missing discriminator

Commit:
reverting changes made to MiSqlConnectionInfo

Commit:
fixing more s360 errors

vasundhraManhas commented:
/azp run

Commit:
fixing s360 correctness issues

vasundhraManhas commented:
/azp run

Commit:
adding $expand parameter in examples of get MI expanded and get VM expanded
